test/unit/lib/otel/consumer.test.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+// Tests to verify that we can map OTEL "consumer" spans to NR segments.
+
+const test = require('node:test')
+const assert = require('node:assert')
+
+const { BasicTracerProvider } = require('@opentelemetry/sdk-trace-base')
+const { SpanKind } = require('@opentelemetry/api')
+
+const { DESTINATIONS } = require('../../../../lib/transaction')
+const helper = require('../../../lib/agent_helper')
+const createSpan = require('./fixtures/span')
+const SegmentSynthesizer = require('../../../../lib/otel/segment-synthesis')
+
+test.beforeEach((ctx) => {
+  const logs = []
+  const logger = {
+    debug(...args) {
+      logs.push(args)
+    }
+  }
+  const agent = helper.loadMockedAgent()
+  const synth = new SegmentSynthesizer(agent, { logger })
+  const tracer = new BasicTracerProvider().getTracer('default')
+
+  ctx.nr = {
+    agent,
+    logger,
+    logs,
+    synth,
+    tracer
+  }
+})
+
+test.afterEach((ctx) => {
+  helper.unloadAgent(ctx.nr.agent)
+})
+
+test('should create consumer segment from otel span', (t) => {
+  const { synth, tracer } = t.nr
+  const span = createSpan({ tracer, kind: SpanKind.CONSUMER })
+  span.setAttribute('messaging.operation', 'receive')
```

Comment:
this test will have to be updated once the feedback is addressed for the new name and also attributes